File: server/datasets/java_source-code/Multiple-D1/organic_3.java
/*
 * Quick Sort - Organic Submission #3
 * Lomuto variant using the first element as pivot.
 */

class Solution {
    public static int partitionFirst(int[] arr, int low, int high) {
        int pivot = arr[low];
        int swapIndex = low;
        for (int i = low + 1; i <= high; i++) {
            if (arr[i] < pivot) {
                swapIndex++;
                int temp = arr[swapIndex];
                arr[swapIndex] = arr[i];
                arr[i] = temp;
            }
        }
        int temp = arr[low];
        arr[low] = arr[swapIndex];
        arr[swapIndex] = temp;
        return swapIndex;
    }

    public static void quickSortFirst(int[] arr, int low, int high) {
        if (low < high) {
            int p = partitionFirst(arr, low, high);
            quickSortFirst(arr, low, p - 1);
            quickSortFirst(arr, p + 1, high);
        }
    }

    public static void main(String[] args) {
        int[] nums = {38, 27, 43, 3, 9, 82, 10};
        quickSortFirst(nums, 0, nums.length - 1);
        System.out.println(java.util.Arrays.toString(nums));
    }
}

File: server/datasets/python_source-code/Multiple-D1/organic_3.py
"""
Quick Sort - Organic Submission #3
Lomuto variant using the first element as pivot.
"""

def partition_first(arr, low, high):
    pivot = arr[low]
    swap_index = low
    for i in range(low + 1, high + 1):
        if arr[i] < pivot:
            swap_index += 1
            arr[swap_index], arr[i] = arr[i], arr[swap_index]
    arr[low], arr[swap_index] = arr[swap_index], arr[low]
    return swap_index

def quick_sort_first(arr, low, high):
    if low < high:
        p = partition_first(arr, low, high)
        quick_sort_first(arr, low, p - 1)
        quick_sort_first(arr, p + 1, high)

if __name__ == "__main__":
    nums = [38, 27, 43, 3, 9, 82, 10]
    quick_sort_first(nums, 0, len(nums) - 1)
    print(nums)
